Library/Sender.ts

```diff
@@ -213,9 +271,15 @@ class Sender {
         // For performance, only run ACL rules if we haven't already during this session
         if (Sender.ACLED_DIRECTORIES[directory] === undefined) {
             // Restrict this directory to only current user and administrator access
-            this._runICACLS(this._getACLArguments(directory), (err) => {
-                Sender.ACLED_DIRECTORIES[directory] = !err;
-                return callback(err);
+            this._getACLIdentity((err, identity) => {
+                if (err) {
+                    return callback(err);
```

Comment:
do we need to set a flag for directory here that will tell that operation didn't succeed and it should not try again?

Reply:
Good catch!